pkg/k8sops/m3db/statefulset.go

```diff
@@ -178,7 +178,7 @@ func NewBaseStatefulSet(ssName, isolationGroup string, cluster *myspec.M3DBClust
 		ObjectMeta: metav1.ObjectMeta{
 			Name:        ssName,
 			Labels:      objLabels,
-			Annotations: objAnnotations,
+			Annotations: annotations.Copy(objAnnotations),
```

Comment:
the sts and the pod were sharing the same annotations map, so when the sts annotations were updated with the operator annotation, the pod would inherit it as well.

Reply:
Interesting - do you think anyone/we may be relying on this behavior even if only accidentally for things like monitoring annotations?

Can probably fix it later with a cluster config update if that’s the case but definitely some kind of behavior i could see someone having started depending on.

Reply:
so I think this currently is a test only bug, since the test generates a stateful set _and_ sets the update annotation for the operator to update it. so the shared annotations are mutated and "corrupted".

in a real env, the annotation is set on an existing sts in k8s, so it doesn't hit this bug.